File: No_1104/Path In Zigzag Labelled Binary Tree.cpp
class Solution {
public:
    vector<int> pathInZigZagTree(int label) {
        int h = log(label)/log(2);
        vector<int> ret;
        ret.push_back(label);
        h--;
        while(h>= 0) {
            label = pow(2, h+1) - (label/2 - pow(2, h)) - 1; //  find parent formula
            ret.insert(ret.begin(), label);
            h--;
        }
        
        return ret;
    }
};

File: No_1104/Path In Zigzag Labelled Binary Tree.py
class Solution:
    def pathInZigZagTree(self, label: int) -> List[int]:
        h = int(log(label)/log(2))
        
        path = [label];
        while h != 0:
            parent = pow(2, h-1) + (pow(2, h) - 1 - int(path[-1] / 2))
            path.append(parent)
            h -= 1
        
        return path[::-1]